shared/src/commonMain/kotlin/org/jetbrains/kotlinconf/ConferenceService.kt

```diff
@@ -50,77 +41,76 @@ val UNKNOWN_SPEAKER: Speaker = Speaker(
 class ConferenceService(
     val context: ApplicationContext,
     val endpoint: String,
-) : CoroutineScope, Closeable {
-    private val storage: ApplicationStorage = ApplicationStorage(context)
-    private var userId2024: String? by storage.bind(String.serializer().nullable) { null }
-    private var needsOnboarding: Boolean by storage.bind(Boolean.serializer()) { true }
-    private var notificationsAllowed: Boolean by storage.bind(Boolean.serializer()) { false }
-    private var _themeStorage: Theme by storage.bind(Theme.serializer()) { Theme.SYSTEM }
-    private val _theme = MutableStateFlow(_themeStorage)
-    val theme: StateFlow<Theme> = _theme.asStateFlow()
+) {
+    private val storage: ApplicationStorage = MultiplatformSettingsStorage(context)
 
     private val client: APIClient by lazy {
         APIClient(endpoint)
     }
 
-    override val coroutineContext: CoroutineContext =
-        SupervisorJob() + Dispatchers.App
+    val scope = CoroutineScope(SupervisorJob() + Dispatchers.Default)
```

Comment:
Could you tell me why do we need to have explicit scope?

Reply:
It's what the recommendation is since https://github.com/Kotlin/kotlinx.coroutines/issues/1581

For me the main benefit is that the scope can be private within the class (which it wasn't here, but fixed in a new commit). It's weird if callers can externally do `service.launch {}` to run something inside the service's scope.

It also allows for the scope to be injected for testing, which we're not doing here, at least not yet, but it gets us one step closer to that.